src/redirections/redirections.c

```diff
@@ -42,18 +42,29 @@ void	redirect_input(t_command *command)
 		infile_redirect(command->infile);
 	else
 	{
-		file = open(command->infile, O_WRONLY | O_CREAT | O_APPEND, 0777);
+		file = open(command->infile, O_WRONLY | O_CREAT | O_APPEND, 0644);
+		if (file == -1)
+			return (EXIT_FAILURE);
 		g_signal_num = EXIT_SUCCESS;
-		while (g_signal_num != SIGINT)
+		while (1)
 		{
 			line = readline("> ");
-			if (is_same_string(line, command->infile))
-				break ;
+			if (!line || is_same_string(line, command->infile))
+				break;
+			if (g_signal_num == SIGINT)
+			{
+				g_signal_num = EXIT_SUCCESS;
```

Comment:
because it needs to be resetted. 
you know how every time we would do << eof followed by ctr^C we would get a different result each time? this resolved it as it was resetting it. 
this whole function is in experimental mode though still, as long as the behaviour of ctrl C is not resolved. so lets keep an eye out.